reana_workflow_controller/rest/utils.py

```diff
@@ -489,23 +489,26 @@ def list_files_recursive_wildcard(directory_path, path):
     ]
 
 
-def download_files_recursive_wildcard(workspace_path, path):
+def download_files_recursive_wildcard(workflow_name, workspace_path, path):
     """Download file(s) matching the given path pattern from the workspace.
 
     This function finds out if the provided pattern corresponds to:
     - a single file; then serves it directly
     - a directory; then packages it into a zip file
     - multiple files; then packages them into a zip file
 
+    :param workflow_name: Full workflow name including run number.
     :param workspace_path: Base workspace directory where files are located.
     :param path: (Wildcard) pattern to use for the download.
     :return: Flask function call to send file to the client.
     """
 
-    def _send_zipped_dir_or_files(dir_path=None, file_paths=None):
+    def _send_zipped_dir_or_files(workflow_name, dir_path=None, file_paths=None):
         """Wrap directory into a zip file in memory and send it to the client."""
         timestr = time.strftime("%Y-%m-%d-%H%M%S")
-        filename = "download_{}.zip".format(timestr)
+        filename = "download_{}_{}_{}.zip".format(
+            workflow_name, remove_upper_level_references(path), timestr
```

Comment:
There's a problem when appending the path. This path can be a wildcard pattern so the generated name can contain special characters causing malformed file names and new directories. Example:

```console
$ reana-client download -w roofit '**/*.C'  
File download_roofit.3_**/*.C_2021-07-20-104455.zip downloaded to /Users/marco/code/reanahub/reana-demo-root6-roofit.
$ tree download_roofit.3_\*\* 
download_roofit.3_**
└── *.C_2021-07-20-104455.zip

0 directories, 1 file
```